frontend/taipy-gui/src/components/Taipy/Metric.tsx

```diff
@@ -50,31 +47,25 @@ const Metric = (props: MetricProps) => {
         width = "100%",
         height
     } = props;
-    const metricValue = useDynamicProperty(props.value, props.defaultValue, 0)
-    const metricThresholdValue = useDynamicProperty(props.threshold, props.defaultThreshold, undefined)
-    const metricDelta = useDynamicProperty(props.delta, props.defaultDelta, undefined)
+    const value = useDynamicProperty(props.value, props.defaultValue, 0)
+    const threshold = useDynamicProperty(props.threshold, props.defaultThreshold, undefined)
+    const delta = useDynamicProperty(props.delta, props.defaultDelta, undefined)
     const className = useClassNames(props.libClassName, props.dynamicClassName, props.className);
     const baseLayout = useDynamicJsonProperty(props.layout, props.defaultLayout || "", emptyLayout);
     const baseStyle = useDynamicJsonProperty(props.style, props.defaultStyle || "", defaultStyle);
 
     const data = useMemo(() => ([
         {
             domain: {x: [0, 1], y: [0, 1]},
-            value: props.format === undefined ? metricValue : parseFloat(sprintf(props.format, metricValue)),
+            value: value,
             type: "indicator",
             mode: (() => {
```

Comment:
why not mode: "gauge+number" + (delta !== undefined ? "+delta" : ""),

Reply:
any way to show a gauge without number ?

Reply:
Yes we can but we might need to add something to check if the user want to not showing the number value. Since number will always be given with a value so we couldn't go with `"number!==undefined"` method. 

I guess we could add a boolean attribute called "displayNumber" so that we could use the mode like this: `"gauge" + (displayNumber ? "+number" : "")  + (delta !== undefined ? "+delta" : "")`

Reply:
it would be something like this:

<|50|metric|display_number=true|>

<img width="356" alt="image" src="https://github.com/Avaiga/taipy/assets/36433663/ab5a8636-c461-4847-b1fb-12986d5bbe9f">